baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/GradleTaskConfigurationAvoidance.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.ERROR,
+        summary = "Forbid gradle apis that eagerly create tasks. Learn more at"
+                + " https://docs.gradle.org/current/userguide/task_configuration_avoidance.html")
+@SuppressWarnings("deprecation")
+public final class GradleTaskConfigurationAvoidance extends BugChecker
+        implements BugChecker.MethodInvocationTreeMatcher {
```

Comment:
does this also work on `.gradle` files written in Groovy?  The tests are only in Java, so I'm not sure if this would run on `.gradle` files.  Or is this only meant to run on gradle plugins written in Java?

Reply:
I believe it does not because error prone only runs on java.